persist/persist_test.go

```diff
@@ -26,3 +26,71 @@ func TestIntegrationRandomSuffix(t *testing.T) {
 		file.Close()
 	}
 }
+
+// TestUnitNewSafeFile checks that a new file is created and that its name is
+// different than the finalName of the new safeFile.
+func TestUnitNewSafeFile(t *testing.T) {
+	// Create safe file.
+	sf, err := NewSafeFile("NewSafeFile test file" + RandomSuffix())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(sf.Name())
+	defer sf.Close()
+
+	// Check that the name of the file is not equal to the final name of the
+	// file.
+	if sf.Name() == sf.finalName {
+		t.Errorf("safeFile temporary filename and finalName are equivalent: %s\n", sf.Name())
+	}
+}
+
+// TestSafeFile tests creating and committing safe files with both relative and
+// absolute paths.
+func TestSafeFile(t *testing.T) {
+	// Generate absolute path filename.
+	absPath := filepath.Join(os.TempDir(), "NewSafeFile test file"+RandomSuffix())
+	// Get relative path filename from absolute path.
+	wd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	relPath, err := filepath.Rel(wd, absPath)
```

Comment:
Also, in this case, because the relative and the absolute path have nothing in common, you are going to end up with the same string.